frontend/src/components/accountPage/UploadProgressModal/UploadProgressContent.tsx

```diff
@@ -0,0 +1,179 @@
+import { FC, useState, useEffect } from 'react';
+import { UserAccountPage } from '../../../utils';
+import { Row, Button, Upload, Typography, Steps, Col, Progress } from 'antd';
+import { UploadOutlined, LoadingOutlined } from '@ant-design/icons';
+import EditableTable from './EditableTable';
+import Papa from 'papaparse';
+const { Text } = Typography;
+const { Step } = Steps;
+export interface IUploadProgressContent {
+  workspaceNamespace: string;
+  setEditing: (value: boolean) => void;
+  stepCurrent: number;
+  stepStatus: 'finish' | 'error' | 'process' | 'wait';
+  setStepCurrent: (value: number) => void;
+  usersCSV: UserAccountPage[];
+  setUsersCSV: (users: UserAccountPage[]) => void;
+  uploadedNumber: number;
+  abortUploading: boolean;
+}
+
+const UploadProgressContent: FC<IUploadProgressContent> = props => {
+  const [fileError, setFileError] = useState<string>('');
+  const [statusProgressBar, setStatusProgessBar] = useState<
+    'active' | 'normal' | 'success' | 'exception'
+  >('active');
+  const { usersCSV, setUsersCSV } = props;
+
+  useEffect(() => {
+    if (props.stepStatus === 'error' || props.abortUploading)
+      setStatusProgessBar('exception');
+    else if (props.uploadedNumber === usersCSV.length)
+      setStatusProgessBar('success');
+    else setStatusProgessBar('active');
+  }, [
+    props.stepStatus,
+    props.uploadedNumber,
+    usersCSV.length,
+    props.abortUploading,
+  ]);
+
+  const onCsvUploaded = (fileInfo: any) => {
+    if (fileInfo.file.status === 'removed') {
+      setUsersCSV([]);
+      return;
+    }
+
+    Papa.parse<any>(fileInfo.file, {
+      header: true,
+      skipEmptyLines: true,
+      complete: (result, _) => {
+        for (const line of result.data) {
+          if (
+            !line['NOME'] ||
+            !(line['COGNOME - (*) Inserito dal docente'] || line['COGNOME']) ||
+            !line['MATRICOLA'] ||
+            !line['EMAIL']
+          ) {
+            setFileError(
+              'Invalid file format, must contain <MATRICOLA, NOME, COGNOME (o COGNOME - (*) Inserito dal docente), EMAIL>'
+            );
+            return;
+          }
+        }
+        const users = result.data.map((user: any, index: Number) => {
+          return {
+            key: user['MATRICOLA'],
+            name: user['NOME'] ?? '',
+            surname:
+              user['COGNOME - (*) Inserito dal docente'] ??
+              user['COGNOME'] ??
+              '',
+            userid: user['MATRICOLA'] ?? '',
+            email: user['EMAIL'] ?? '',
+            currentRole: 'user',
+            workspaces: [{ name: props.workspaceNamespace, role: 'user' }],
+          };
+        });
```

Comment:
```suggestion
        const users = result.data.map((user: any, index: Number) => ({
          key: user['MATRICOLA'],
          name: user['NOME'] ?? '',
          surname:
            user['COGNOME - (*) Inserito dal docente'] ??
            user['COGNOME'] ??
            '',
          userid: user['MATRICOLA'] ?? '',
          email: user['EMAIL'] ?? '',
          currentRole: 'user',
          workspaces: [{ name: props.workspaceNamespace, role: 'user' }],
        }));
```
Also check since there should be an enum in generatedTypes with the role that you can use already, instead of using 'user' and 'manager'.